mem0/memory/main.py

```diff
@@ -708,16 +711,19 @@ def _create_procedural_memory(self, messages, metadata=None, prompt=None):
 
         return result
 
-    def _update_memory(self, memory_id, data, existing_embeddings, metadata=None):
+    def _update_memory(self, memory_id, data, existing_embeddings, metadata: dict | None = None):
```

Comment:
Change this to `Optional[dict]` because this project supports Python >=3.9 and `|` was introduced in Python 3.10